src/libpcp/variable/memory/mod.rs

```diff
@@ -246,7 +243,7 @@ mod test {
         fn from_shape(&mut self, depth: usize, shape: &TreeShape) -> usize {
             use self::TreeShape::*;
             match shape {
-                &Node(ref value, ref children) if depth > 0 => {
+                Node(value, children) if depth > 0 => {
```

Comment:
I see `ref` was removed in many places, is it by reference by default now? Instead of being by copy?

Reply:
This changed some years ago as part of match ergonomics: https://rust-lang.github.io/rfcs/2005-match-ergonomics.html

Reply:
My Rust knowledge is completely outdated :sweat: Thanks.